File: app/src/main/java/com/icostel/arhitecturesample/navigation/AppScreenProvider.java
package com.icostel.arhitecturesample.navigation;

import com.icostel.arhitecturesample.ui.loginuser.LoginUserActivity;
import com.icostel.arhitecturesample.ui.main.MainActivity;
import com.icostel.arhitecturesample.ui.newuser.NewUserActivity;
import com.icostel.arhitecturesample.ui.userdetails.UserDetailsActivity;
import com.icostel.commons.navigation.ScreenProvider;

import javax.inject.Inject;

public class AppScreenProvider extends ScreenProvider {

    public static final int LOGIN_USER = 1;
    public static final int USER_DETAILS = 2;
    public static final int NEW_USER = 4;
    public static final int MAIN = 5;


    @Inject
    AppScreenProvider() {
        super();
        initScreenMap();
    }

    @Override
    protected void initScreenMap() {
        SCREEN_MAP.put(LOGIN_USER, LoginUserActivity.class);
        SCREEN_MAP.put(NEW_USER, NewUserActivity.class);
        SCREEN_MAP.put(MAIN, MainActivity.class);
        SCREEN_MAP.put(USER_DETAILS, UserDetailsActivity.class);
    }
}

File: app/src/main/java/com/icostel/arhitecturesample/navigation/AppScreenProvider.kt
package com.icostel.arhitecturesample.navigation

import com.icostel.arhitecturesample.ui.screens.loginuser.LoginUserActivity
import com.icostel.arhitecturesample.ui.screens.main.MainActivity
import com.icostel.arhitecturesample.ui.screens.newuser.NewUserActivity
import com.icostel.arhitecturesample.ui.screens.userdetails.UserDetailsActivity
import com.icostel.commons.navigation.ScreenProvider

import javax.inject.Inject
import javax.inject.Singleton

@Singleton
class AppScreenProvider @Inject
internal constructor() : ScreenProvider() {

    init {
        initScreenMap()
    }

    override fun initScreenMap() {
        SCREEN_MAP.put(LOGIN_USER, LoginUserActivity::class.java)
        SCREEN_MAP.put(NEW_USER, NewUserActivity::class.java)
        SCREEN_MAP.put(MAIN, MainActivity::class.java)
        SCREEN_MAP.put(USER_DETAILS, UserDetailsActivity::class.java)
    }

    companion object {

        const val LOGIN_USER = 1
        const val USER_DETAILS = 2
        const val NEW_USER = 4
        const val MAIN = 5
    }
}
